QueueUsingStack.java

```diff
@@ -0,0 +1,71 @@
+import java.util.Stack;
+
+public class QueueUsingStack {
+	Stack<Integer> stack1; // reference stack
+    Stack<Integer> stack2; // queue stack
+    
+    /** Initialize your data structure here. */
+    public QueueUsingStack() {
+       stack1 = new Stack<>();
+        stack2 = new Stack<>();
+    }
+    
+    /** Push element x to the back of queue. */
+    
+    // if queue stack is empty, push value in it.
+    // otherwise, transfer the values from queue stack (until its not empty) to reference stack, push new value 
+    // in queue stack and transfer values from reference stack to queue stack.
```

Comment:
Your explanation to this is awesome!!

Reply:
O(n) because as soon as we push any element, we move all the elements in the other stack.